sdk/storage/Azure.Storage.DataMovement/src/DataTransferStatus.cs

```diff
@@ -105,15 +104,48 @@ internal bool TrySetTransferStateChange(DataTransferState state)
             return Interlocked.Exchange(ref _stateValue, (int)state) != (int)state;
         }
 
-        /// <summary>
-        /// Indicates whether the current object is equal to another object of the same type.
-        /// </summary>
-        /// <param name="other">An object to compare with this object.</param>
-        /// <returns>Returns true if the current object is equal to the other parameter; otherwise, false.</returns>
+        /// <inheritdoc/>
         public bool Equals(DataTransferStatus other)
-            => State.Equals(other.State) &&
-            HasFailedItems.Equals(other.HasFailedItems) &&
-            HasSkippedItems.Equals(other.HasSkippedItems);
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            return State.Equals(other.State) &&
+                HasFailedItems.Equals(other.HasFailedItems) &&
+                HasSkippedItems.Equals(other.HasSkippedItems);
+        }
+
+        /// <inheritdoc/>
+        public static bool operator ==(DataTransferStatus left, DataTransferStatus right)
+        {
+            if (left is null)
+            {
+                if (right is null)
+                {
+                    return true;
+                }
+                return false;
+            }
+            return left.Equals(right);
```

Comment:
style but perhaps we avoid nested ifs and imrpove clarity of what is being tested in those initial returns. perhaps something like the following:
```csharp
if (left is null != right is null)
{
    return false;
}
return left?.Equals(right) ?? true;
```

Reply:
Sure, copied this from the .NET developer guide on equality but this looks a bit nicer admittedly.